optimum/intel/pipelines/pipeline_base.py

```diff
@@ -349,128 +140,138 @@ def pipeline(
+            pipeline will be allocated.
+        device_map (`str` or `Dict[str, Union[int, str, torch.device]]`, *optional*):
+            Sent directly as `model_kwargs` (just a simpler shortcut). When `accelerate` library is present, set
+            `device_map="auto"` to compute the most optimized `device_map` automatically (see
+            [here](https://huggingface.co/docs/accelerate/main/en/package_reference/big_modeling#accelerate.cpu_offload)
+            for more information).
+
+            <Tip warning={true}>
+
+            Do not use `device_map` AND `device` at the same time as they will conflict
+
+            </Tip>
+
         torch_dtype (`str` or `torch.dtype`, *optional*):
             Sent directly as `model_kwargs` (just a simpler shortcut) to use the available precision for this model
             (`torch.float16`, `torch.bfloat16`, ... or `"auto"`).
+        trust_remote_code (`bool`, *optional*, defaults to `False`):
+            Whether or not to allow for custom code defined on the Hub in their own modeling, configuration,
+            tokenization or even pipeline files. This option should only be set to `True` for repositories you trust
+            and in which you have read the code, as it will execute code present on the Hub on your local machine.
         model_kwargs (`Dict[str, Any]`, *optional*):
             Additional dictionary of keyword arguments passed along to the model's `from_pretrained(...,
             **model_kwargs)` function.
+        kwargs (`Dict[str, Any]`, *optional*):
+            Additional keyword arguments passed along to the specific pipeline init (see the documentation for the
+            corresponding pipeline class for possible values).
 
     Returns:
         [`Pipeline`]: A suitable pipeline for the task.
 
     Examples:
-
     ```python
-    >>> import torch
-    >>> from optimum.intel.pipelines import pipeline
-
-    >>> pipe = pipeline('text-generation', 'gpt2', torch_dtype=torch.bfloat16)
-    >>> pipe("Describe a real-world application of AI in sustainable energy.")
-    ```"""
-
-    if model_kwargs is None:
-        model_kwargs = {}
-
-    commit_hash = kwargs.pop("_commit_hash", None)
-
-    if task is None and model is None:
-        raise RuntimeError(
-            "Impossible to instantiate a pipeline without either a task or a model "
-            "being specified. "
-            "Please provide a task class or a model"
-        )
-
-    if model is None and tokenizer is not None:
-        raise RuntimeError(
-            "Impossible to instantiate a pipeline with tokenizer specified but not the model as the provided tokenizer"
-            " may not be compatible with the default model. Please provide a PreTrainedModel class or a"
-            " path/identifier to a pretrained model when providing tokenizer."
-        )
+    >>> from optimum.intel import pipeline
```

Comment:
Does it really work this way? I don't see `pipeline` being exposed at `optimum.intel.__init__`

Reply:
I just copied directly from transformers and adapted ! it should be added to `optimum.intel.__init__` then !